src/index.ts

```diff
@@ -34,14 +33,36 @@ export enum Status {
   NotAcceptable = 406
 }
 
+export interface CookieOptions {
+  expires?: Date
+  domain?: string
+  httpOnly?: boolean
+  maxAge?: number
+  path?: string
+  sameSite?: boolean | 'strict' | 'lax'
+  secure?: boolean
+  signed?: boolean
+}
+
 /**
  * A `Conn`, short for "connection", models the entirety of a connection between the HTTP server and the user agent,
  * both request and response.
  * State changes are tracked by the phanton type `S`
  */
-export class Conn<S> {
+export interface Conn<S> {
   readonly '-S': S
-  constructor(readonly req: express.Request, readonly res: express.Response) {}
+
+  endResponse: () => void
+
+  getBody: () => any
+  getHeader: (name: string) => string | undefined
+  getParams: () => any
+  getQuery: () => any
+
+  setBody: (body: any) => void
```

Comment:
can we narrow down this type?

```ts
setBody: (body: any) => void
```

`string`? `t.mixed`?

Reply:
I've added a new `Body` type to cover the overlap between koa and express. It means you lose the ability to send a `Stream` if you are using koa though. Express doesn't support that I think.